repo/godep.go

```diff
@@ -0,0 +1,100 @@
+/* Copyright 2019 The Bazel Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repo
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"sync"
+
+	"github.com/bazelbuild/bazel-gazelle/label"
+)
+
+type goDepLockFile struct {
+	ImportPath   string
+	GoVersion    string
+	GodepVersion string
+	Packages     []string
+	Deps         []goDepProject
+}
+
+type goDepProject struct {
+	ImportPath string
+	Rev        string
+}
+
+func importRepoRulesGoDep(filename string, cache *RemoteCache) ([]Repo, error) {
+	data, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	file := goDepLockFile{}
+
+	if err := json.Unmarshal(data, &file); err != nil {
+		return nil, err
+	}
+
+	var wg sync.WaitGroup
+	var updateLock sync.Mutex
```

Comment:
Rather than keeping a `map[string]string` and an `*error` guarded by a mutex, I think this would be simpler:

```
roots := make([]string, len(file.Deps))
errs := make([]error, len(file.Deps))

...

for i, p := range file.Deps {
  go func(i int, p goDepProject) {
    roots[i], _, errs[i] = cache.Root(p.ImportPath)
    wg.Done()
  }(i, p)
}
```

`roots` can be a slice instead of a map: when we iterate through the packages in the loop at the end, each package will have a root at the corresponding index.

`errs` should be a slice to avoid a race on `*err`. That same loop at the end can check for an error at each index and return if it's non-nil.